File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/ui/map/ETAWindowAdapter.java
package rectangledbmi.com.pittsburghrealtimetracker.ui.map;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.Marker;

import rectangledbmi.com.pittsburghrealtimetracker.R;

/**
 * <p>This is the adapter class for the marker option window.</p>
 *
 * Created by epicstar on 12/22/14.
 */
public class ETAWindowAdapter implements GoogleMap.InfoWindowAdapter {

    private LayoutInflater inflater;
    private View infoWindow;

    public ETAWindowAdapter(LayoutInflater inflater) {
        this.inflater = inflater;
    }

    @Override
    public View getInfoWindow(Marker marker) {
        return null; //should always return null
    }

    @Override
    public View getInfoContents(Marker marker) {
        if(infoWindow == null)
            infoWindow = inflater.inflate(R.layout.map_eta_infowindow, null);

        TextView title = (TextView) infoWindow.findViewById(R.id.title);
        TextView snippet = (TextView) infoWindow.findViewById(R.id.snippet);
        title.setText(marker.getTitle());
        snippet.setText(marker.getSnippet());

        return infoWindow;
    }
}

File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/ui/map/ETAWindowAdapter.kt
package rectangledbmi.com.pittsburghrealtimetracker.ui.map

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.widget.TextView

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker

import rectangledbmi.com.pittsburghrealtimetracker.R

/**
 *
 * This is the adapter class for the marker option window.
 *
 * Created by epicstar on 12/22/14.
 */
class ETAWindowAdapter(private val inflater: LayoutInflater) : GoogleMap.InfoWindowAdapter {
    private var infoWindow: View? = null

    override fun getInfoWindow(marker: Marker): View? {
        return null //should always return null
    }

    @SuppressLint("InflateParams")
    override fun getInfoContents(marker: Marker): View {
        val infoWindow = infoWindow ?: inflater.inflate(R.layout.map_eta_infowindow, null)

        val title = infoWindow.findViewById<View>(R.id.title) as TextView
        val snippet = infoWindow.findViewById<View>(R.id.snippet) as TextView
        title.text = marker.title
        snippet.text = marker.snippet

        return infoWindow as View
    }
}
